### ISSUE TITLE
Lag when moving past tiles with many items

### ISSUE THREAD

**Issue by Funguss** [CleverRaven/Cataclysm-DDA#38340]
It seems that having many items in a tile causes lag when moving past it. This is unrelated to #37554  as there aren't any maps in my piles. I'm using 4x monster spawns and am trying to keep things tidy, but massive piles of dirty clothes and misc loot are causing me to hoard (as I don't want to even approach them!). I'm half inclined to think part of this is my rig being slow, but I doubt that's the full story, or even part of it.

# Steps To Reproduce

1. Dump thousands of litres of small junk items in one area.
2. Try walking about in it.

# Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-12504-g9015618 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Makeshift Items Mod [makeshift],
    More Survival Tools [More_Survival_Tools],
    Mythological Replicas [nw_pack],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    No Rail Stations [No_Rail_Stations],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Roadheader and other mining vehicles [Heavy miners],
    No Fungal Monsters [No_Fungi],
    Beta National Guard Camp [national_guard_camp]
]

**Comment by ZhilkinSerg** [comment]
Do you have a savegame affected by this issue?

**Comment by Funguss** [comment]
Sure thing, should have done that in the first place, shouldn't I? 
[Point Blank.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4253829/Point.Blank.zip)

**Comment by ZhilkinSerg** [comment]
Where is that pile in question?

**Comment by Funguss** [comment]
Character is standing in the middle of them, every pile generates lag, but the more items the worse it is. It seems to be related to the amount of pages there are. Walking east and west between those two central tiles should be noticeably slow, walking east especially. It takes about two seconds for me.

There's a fire ring in the centre of the zoo, too, with filthy armour and clothing nearby. West of that are gigantic pliles of potential laundry. All of these cause lag when walked past, too, especially the western laundry pile. The central tile has 41 pages of items bordering them, according to adanced inventory management.

**Comment by foulman** [comment]
I notice this behavior all the time in my deathmobile storage areas. The more you have the slower it gets, I always chalked it up to something like temperature calculations for individual rags or something.

**Comment by Funguss** [comment]
This behaviour particularly? I doubt it's temperature calculations, as it seems odd to process those when the PC walks past them. I also think it's related to the amount of item pages, not the amount of individual items. My filthy clothing piles have no stacking items, occupy fewer tiles and have many less items overall than my main loot pile, yet has equivalent pages and lag.

I'm wondering if the advanced inventory is buffering as you move, but this occurs even if 'a'll isn't selected on either column. The inventory lists for each tile may still be getting built, though. The amount of lag on move does seem equivalent to other item commands, so I imagine a list is being built for some reason.

**Comment by kevingranade** [comment]
My guess is that this is auto pick up kicking in and iterating over all the
items in the stack trying to decide whether to pick any of them up.

**Comment by Funguss** [comment]
Yes!! Very nicely done! Turned autopickup off and it's no longer an issue. I didn't even realise I had it on, it's bound to "unknown key 7", whatever that is. I must have hit it by accident at some point.

Ok, should I close this then or update it for autopickup? A quick search shows there's no open issue at the moment and it seems it could do with some optimization.

**Comment by kevingranade** [comment]
Leave it open, it needs optimization.

**Comment by kevingranade** [comment]
Can you provide your autopickup configuration? I'm not having any luck getting this to trigger.

**Comment by Funguss** [comment]
Sure. It was "wheat", disabled, and "*arrow", disabled. So having autopickup on without anything defined, basically.
